### PR TITLE
feat(@ciscospark/i-p-team): handle teamRoomStatus decryption

### PR DESCRIPTION
**Pull request by aimeex3**
## Description

There is a new activity event with teamRoomStatus object type. https://sqbu-github.cisco.com/gist/brialam/6702e04c04f59fd97526b0fb3caaee36 This PR handles the decryption of this new objectType

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] create team with user A and user B, create team space with user A. User A changes the display name for the teams space. User B should receive the teamRoomStatus activity with the display name decrypted

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@ciscospark/internal-plugin-team/test/integration/spec/mercury.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+      }));
+
+    after(() => Promise.all([
+      kirk && kirk.spark.internal.mercury.disconnect(),
+      spock && spock.spark.internal.mercury.disconnect()
+    ]));
+
+    describe('when mercury event is received', () => {
+      let teamConvo, team;
+      before(() => kirk.spark.internal.team.create({
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/internal-plugin-team/test/integration/spec/mercury.js]: let's break these up into separate named before hooks. It makes it a lot easier to see where things are going wrong when we know what action timed out.

File: packages/node_modules/@ciscospark/internal-plugin-team/test/integration/spec/mercury.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          teamConvo = c;
+        }));
+
+      it('updates the displayName of an unjoined team convo', () => {
+        const update = {
+          displayName: `updated-team-convo--${uuid.v4()}`,
+          objectType: 'conversation'
+        };
+
+        const activityChecker = (activity) => activity.verb === 'update' && activity.object && activity.object.objectType === 'teamRoomStatus';
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/internal-plugin-team/test/integration/spec/mercury.js]: try using `get()` from lodash to make the `activity.object.objectType...` bits easier to read

File: packages/node_modules/@ciscospark/test-helper-mocha/src/index.js [hunk 3]
```diff
@@ -147,6 +147,34 @@ module.exports = {
...
+  expectActivity: function expectActivity(max, event, emitter, activityChecker, msg) {
+    let timer;
+    return Promise.race([
+      new Promise((resolve, reject) => {
+        timer = setTimeout(() => {
+          reject(new Error(`${event} did not fire within ${max}ms${msg ? `: ${msg}` : ''}`));
+        }, max);
+      }),
+      new Promise((resolve) => {
+        emitter.on(event, (result) => {
```
Comment (ianwremmel) [inline packages/node_modules/@ciscospark/test-helper-mocha/src/index.js]: take a look at `handleErrorEvent`. using `emitter.on` without unbinding the handler could retain the emitter longer than we want and cause weird things to happen in other tests. `handleErrorEvent` dealt with a similar need but unbinds at the end.

Comment (aimeex3) [reply packages/node_modules/@ciscospark/test-helper-mocha/src/index.js]: 👍 added unbind to the end

**Pull request merged**